massdash/server/FeatureMapPlotterServer.py

```diff
@@ -0,0 +1,206 @@
+from ..ui.ChromatogramPlotUISettings import ChromatogramPlotUISettings
+from ..ui.PeakPickingUISettings import PeakPickingUISettings
+
+class FeatureMapPlotterServer:
+    """
+    A class that generates plots from FeatureMaps
+
+    Args:
+        feature_map_dict (dict): A dictionary containing transition groups.
+        transition_list_ui (TransitionListUISettings): An object representing the transition list UI.
+        chrom_plot_settings (ChromatogramPlotUISettings): An object representing the chromatogram plot settings.
+
+    Attributes:
+        feature_map_dict (dict): A dictionary containing transition groups.
+        transition_list_ui (object): An object representing the transition list UI.
+        chrom_plot_settings (ChromatogramPlotUISettings): An object representing the chromatogram plot settings.
+        plot_obj_dict (dict): A dictionary containing the generated plot objects.
+
+    Methods:
+        generate_chromatogram_plots: Generates chromatogram plots for each file in the transition group dictionary.
+        _get_plot_settings: Returns the plot settings dictionary for a given plot type.
+        _generate_plot: Generates a plot object for a given transition group and plot settings.
+
+    """
+
+    def __init__(self, 
+                 featureMapCollection: FeatureMapCollection,
+                 chrom_loader: GenericChromatogramLoader,
+                 transition_list_ui: TransitionListUISettings, 
+                 chrom_plot_settings: ChromatogramPlotUISettings, 
+                 peak_picking_settings: PeakPickingUISettings, 
+                 verbose: bool=False):
+        self.featureMapCollection = featureMapCollection
+        self.chrom_loader = chrom_loader
+        self.transition_list_ui = transition_list_ui
+        self.chrom_plot_settings = chrom_plot_settings
+        self.peak_picking_settings = peak_picking_settings
+        self.plot_obj_dict = defaultdict(list) # a dictionary of plot objects with the mapping <RunName>:[<PlotObject>] (multiple plots per run allowed)
+        self.verbose = verbose
+
+
+    def generate_plots(self):
+        """
+        Generates plots based on user settings
+
+        Returns:
+            None
+        """
+        if self.chrom_plot_settings.display_plot_dimension_type == '1D':
+            self.generate_one_dimensional_plots()
+        elif self.chrom_plot_settings.display_plot_dimension_type == '2D':
+            self.generate_two_dimensional_plots()
+        elif self.chrom_plot_settings.display_plot_dimension_type == '3D':
+            self.generate_three_dimensional_plots()
+        else:
+            raise ValueError(f"Invalid plot dimension: {self.chrom_plot_settings.plot_dimension}. Valid options are '1D', '2D', and '3D'.")
+
+        return self
+
+    def generate_one_dimensional_plots(self):
+        """
+        Generates plots for each file in the transition group dictionary.
+        """
+
+
+        if self.chrom_plot_settings.display_spectrum:
```

Comment:
When the display spectrum button is pressed, it doesn't actually display the spectrum plot for some reason?

![image](https://github.com/Roestlab/massdash/assets/32938975/3aebefb7-b90d-4fed-a8f8-35855a54a51d)

When deselecting display chromatogram, it throws an error because there is nothing to display even when spectrum is selected

![image](https://github.com/Roestlab/massdash/assets/32938975/79f81c64-fa05-4b41-a752-954fa46d7a8f)

Same thing when deselecting all plots

![image](https://github.com/Roestlab/massdash/assets/32938975/6e6dd8eb-154d-48b0-8729-bbe0f23ea741)

Reply:
I for some reason am still running into the issue of the spectrum plot not being displayed if the Display spectrum checkbox is toggled on.

Reply:
![image](https://github.com/Roestlab/massdash/assets/32938975/4ea3647f-1796-43d1-8cf6-c8bc0bbc398a)